addons/stock/models/res_config_settings.py

```diff
@@ -82,4 +84,19 @@ def set_values(self):
                 ('show_operations', '=', False)
             ])
             picking_types.sudo().write({'show_operations': True})
+
+        # Deactivate putaway rules with storage category when not in storage category
+        # group. Otherwise, active them.
+        PutawayRule = self.env['stock.putaway.rule']
```

Comment:
Activate (or deactivate) the putaway rule only if the set_value() is related to `group_stock_storage_categories`. RYV did something similar here: 
https://github.com/odoo/odoo/blob/b4ac583f852d63f0a7ab1afa0b7562272751242c/addons/stock/models/res_config_settings.py#L56-L64
where he save the current state before calling `super`